code/game/gamemodes/gang/dominator.dm

```diff
@@ -0,0 +1,225 @@
+	..()
+	if(operating == -1)
+		user << "<span class='danger'>It looks completely busted.</span>"
+		return
+
+	var/time
+	if(gang && isnum(gang.dom_timer))
+		time = max(gang.dom_timer, 0)
+		if(time > 0)
+			user << "<span class='notice'>Hostile Takeover in progress. Estimated [time] seconds remain.</span>"
+		else
+			user << "<span class='notice'>Hostile Takeover of [station_name()] successful. Have a great day.</span>"
+	else
+		user << "<span class='notice'>System on standby.</span>"
+	user << "<span class='danger'>System Integrity: [round((health/maxhealth)*100,1)]%</span>"
+
+/obj/machinery/dominator/process()
+	..()
+	if(gang && isnum(gang.dom_timer))
+		if(gang.dom_timer > 0)
+			playsound(loc, 'sound/items/timer.ogg', 10, 0)
+			if(!warned && (gang.dom_timer < 180))
+				warned = 1
+				var/area/domloc = get_area(loc)
+				gang.message_gangtools("Less than 3 minutes remain in hostile takeover. Defend your dominator at [initial(domloc.name)]!")
+				for(var/datum/gang/G in ticker.mode.gangs)
+					if(G != gang)
+						G.message_gangtools("WARNING: [gang.name] Gang takeover imminent. Their dominator at [initial(domloc.name)] must be destroyed!",1,1)
+		else
+			processing_objects.Add(src)
+
+/obj/machinery/dominator/proc/healthcheck(damage)
+	var/iconname = "dominator"
+	if(gang)
+		iconname += "-[gang.color]"
+		luminosity = 3
+
+	var/datum/effect/system/spark_spread/sparks = new /datum/effect/system/spark_spread
+
+	health -= damage
+
+	if(health > (maxhealth/2))
+		if(prob(damage*2))
+			sparks.set_up(5, 1, src)
+			sparks.start()
+	else if(operating >= 0)
+		sparks.set_up(5, 1, src)
+		sparks.start()
+		overlays += "damage"
+
+	if(operating != -1)
+		if(health <= 0)
+			set_broken()
+		else
+			icon_state = iconname
+
+	if(health <= -100)
+		new /obj/item/stack/sheet/plasteel(src.loc)
+		qdel(src)
+
+/obj/machinery/dominator/proc/set_broken()
+	if(gang)
+		gang.dom_timer = "OFFLINE"
+
+		var/takeover_in_progress = 0
+		for(var/datum/gang/G in ticker.mode.gangs)
+			if(isnum(G.dom_timer))
+				takeover_in_progress = 1
+				break
+		if(!takeover_in_progress)
+			shuttle_master.emergencyNoEscape = 0
+			if(shuttle_master.emergency.mode == SHUTTLE_STRANDED)
+				shuttle_master.emergency.mode = SHUTTLE_DOCKED
+				shuttle_master.emergency.timer = world.time
+				command_announcement.Announce("Hostile enviroment resolved. You have 3 minutes to board the Emergency Shuttle.", null, 'sound/AI/shuttledock.ogg', "Priority")
+			else
+				command_announcement.Announce("All hostile activity within station systems have ceased.","Network Alert")
+
+			if(get_security_level() == "delta")
+				set_security_level("red")
+
+		gang.message_gangtools("Hostile takeover cancelled: Dominator is no longer operational.[gang.dom_attempts ? " You have [gang.dom_attempts] attempt remaining." : " The station network will have likely blocked any more attempts by us."]",1,1)
+
+	luminosity = 0
+	icon_state = "dominator-broken"
```

Comment:
* Should be moved into an `update_icon` override.
* In the current state, if `healthcheck()` were called after this got set the sprite would reset.